streams/src/main/java/org/apache/kafka/streams/processor/internals/TaskManager.java

```diff
@@ -325,6 +325,31 @@ private void closeDirtyAndRevive(final Collection<Task> taskWithChangelogs, fina
         }
     }
 
+    private Map<Task, Set<TopicPartition>> assignInitialTasks(final Map<TaskId, Set<TopicPartition>> tasksToAssign,
+                                                              final String threadLogPrefix,
+                                                              final TopologyMetadata topologyMetadata,
+                                                              final ChangelogRegister changelogReader) {
+        if (stateDirectory.hasInitialTasks()) {
+            final Map<Task, Set<TopicPartition>> assignedTasks = new HashMap<>(tasksToAssign.size());
+            for (final Map.Entry<TaskId, Set<TopicPartition>> entry : tasksToAssign.entrySet()) {
+                final TaskId taskId = entry.getKey();
+                final Set<TopicPartition> inputPartitions = entry.getValue();
+                final Task task = stateDirectory.assignInitialTask(taskId);
```

Comment:
Seems `assignInitialTask` does not really do the assignment, but we do the actual assignment here? Should we rename the method to `removedInitializedTask()` instead?